posthog-core/src/patch.ts

```diff
@@ -0,0 +1,46 @@
+// import { patch } from 'rrweb/typings/utils'
+// copied from: https://github.com/PostHog/posthog-js/blob/main/src/extensions/replay/rrweb-plugins/patch.ts
+// which was copied from https://github.com/rrweb-io/rrweb/blob/8aea5b00a4dfe5a6f59bd2ae72bb624f45e51e81/packages/rrweb/src/utils.ts#L129
+// which was copied from https://github.com/getsentry/sentry-javascript/blob/b2109071975af8bf0316d3b5b38f519bdaf5dc15/packages/utils/src/object.ts
+import { isFunction } from './utils'
+
+export function patch(
+  source: { [key: string]: any },
+  name: string,
+  replacement: (...args: unknown[]) => unknown
+): () => void {
+  try {
+    if (!(name in source)) {
+      return () => {
+        //
+      }
```

Comment:
from a bundle size perspective i wonder how many times we have `no-op` as a bare declaration
if it's more than just this two we might get a saving from having noOp as a global 🤔

Reply:
don't block here though